pennylane/pulse/rydberg_hamiltonian.py

```diff
@@ -0,0 +1,236 @@
+individual (trapped) rydberg atoms under the excitation of several laser fields."""
+import warnings
+from dataclasses import dataclass
+from typing import Callable, List, Union
+
+import numpy as np
+
+import pennylane as qml
+from pennylane.ops import SProd
+from pennylane.wires import Wires
+
+from .parametrized_hamiltonian import ParametrizedHamiltonian
+
+
+def rydberg_interaction(register: list, wires=None, interaction_coeff: float = 862690 * 2 * np.pi):
+    r"""Returns a :class:`ParametrizedHamiltonian` representing the interaction of an ensemble of
+    Rydberg atoms due to the Rydberg blockade
+
+    .. math::
+
+        \sum_{i<j} V_{ij} n_i n_j
+
+    where :math:`n_i` corresponds to the projector on the Rydberg state of the atom :math:`i`, and
+    :math:`V_{ij}` is the van der Waals potential:
+
+    .. math::
+
+        V_{ij} = \frac{C_6}{R_{ij}^6}
+
+    where :math:`R_{ij}` is the distance between the atoms :math:`i` and :math:`j`, and :math:`C_6`
+    is the Rydberg interaction constant, which defaults to :math:`862690 \times 2\pi MHz \times \mu m^6`.
+
+    Args:
+        register (list): list of coordinates of the Rydberg atoms (in micrometers)
+        wires (list): List of wires containing the wire values for all the atoms. This list should
+            have the same length as ``register``. If ``None``, each atom's wire value will
+            correspond to its index in the ``register`` list.
+        interaction_coeff (float): Rydberg interaction constant in units: :math:`\text{MHz} \times \mu \text{m}^6`.
+            Defaults to :math:`862690 \times 2\pi \text{ MHz} \times \mu \text{m}^6`.
+
+    Returns:
+        RydbergHamiltonian: a :class:`~.ParametrizedHamiltonian` representing the atom interaction
+    """
+    if wires is None:
+        wires = list(range(len(register)))
+    elif len(wires) != len(register):
+        raise ValueError("The length of the wires and the register must match.")
+
+    def rydberg_projector(wire: int) -> SProd:
+        """Returns the projector into the Rydberg state for the given wire.
+
+        Args:
+            wire (int): wire value
+
+        Returns:
+            SProd: projector into the Rydberg state
+        """
+        return qml.s_prod(1 / 2, (1 - qml.PauliZ(wire)))
+
+    coeffs = []
+    observables = []
+    for idx, (pos1, wire1) in enumerate(zip(register[:-1], wires[:-1])):
+        for pos2, wire2 in zip(register[(idx + 1) :], wires[(idx + 1) :]):
+            atom_distance = np.linalg.norm(qml.math.array(pos2) - pos1)
+            Vij = interaction_coeff / (abs(atom_distance) ** 6)  # van der Waals potential
+            coeffs.append(Vij)
+            observables.append(qml.prod(rydberg_projector(wire1), rydberg_projector(wire2)))
+
+    return RydbergHamiltonian(
+        coeffs, observables, register=register, interaction_coeff=interaction_coeff
+    )
+
+
+def rydberg_transition(rabi, detuning, phase, wires):
```

Comment:
How set are we on the name? Could this be `rydberg_drive` instead of `rydberg_transition`? 

I'm putting together a notebook that simulates a couple experiments they have in the Aquila examples, using the PL interface. To do that, I'm using this to add an H_shift term that is only dependent on local detuning, i.e. `rabi=0` and this term does not drive the Rydberg transition. Here it seems like `rydberg_drive` would be a more all-encompassing name, because it includes all drive defined relative to the Rydberg frequency, even if its not driving the transition.

Reply:
@trbromley